internal/renderer/renderer.go

```diff
@@ -346,13 +362,20 @@ func textWithNewlineReplacement(node ast.Node, replacement []byte) []byte {
 	if node, ok := node.(*ast.Link); ok && node.Footnote != nil {
 		fmt.Fprintf(&buf, "[^%d]", node.NoteID)
 	}
-	if node := node.AsLeaf(); node != nil {
+	if leaf := node.AsLeaf(); leaf != nil {
 		// replace all newlines in text with preferred symbols; this may
 		// be spaces for general text, allowing for soft wrapping, which
 		// is recommended as per Gemini spec p. 5.4.1, or line breaks
 		// with a blockquote symbols for blockquotes, or just nothing
 		buf.Write(delimiter)
-		buf.Write(lineBreakCharacters.ReplaceAll(node.Literal, replacement))
+		switch node.(type) {
+		case *ast.Hardbreak:
```

Comment:
When does this appear in the AST? `(Renderer).htmlBlock/subscript/superscript` seem to already handle the `<br>` tag if it ever appears.

Reply:
gomarkdown is supposed to generate a hard break if two spaces precedes a newline, but it looks like I didn't do enough testing! https://github.com/gomarkdown/markdown/blob/2399b65079f5351c32f5339701cad940ad077d0d/parser/inline.go#L169

There's also a parser extension to replace every newline with a hard break but we probably don't want that one :)

I actually didn't know about this Markdown feature until now.